src/controllers/generate-token.ts

```diff
@@ -3,18 +3,18 @@ import { TokenGenerator } from '../components/token-generator'
 import { UserStorage } from '../components/storage/types'
 
 export function generateToken(
-    {tokenGenerator, AwsUserStorage}:
-    {tokenGenerator: TokenGenerator, AwsUserStorage: UserStorage}
+    {tokenGenerator, userStorage}:
+    {tokenGenerator: TokenGenerator, userStorage: UserStorage}
 ) {
     return async function handleGenerateTokenRequest({installTime}) {
         let id = tokenGenerator.generateToken()
-
-        while(!(await AwsUserStorage.isUserById(id))) {
+        while((await userStorage.userExists(id))) {
             id = tokenGenerator.generateToken()
         }
 
-        await AwsUserStorage.storeUser(id, installTime)
+        const user = await userStorage.storeUser(id, installTime)
```

Comment:
Rejections are never handled from this Promise, so if this endpoint is called and there is a problem storing the new user and `storeUser` rejects, I'm not sure if a response would ever be sent (or maybe there is a global error handler somewhere that handles it? This is a valid way of handling misc. errors in express). 
Same issue with `eventLog` and `uninstall` endpoints

Reply:
Yes, or look into error handling in express. I know from past experience there is smart ways of handling general errors on all routes (global error handling), which may be appropriate here.